src/main/java/com/group1/stagesWs/service/OffreService.java

```diff
@@ -31,4 +32,18 @@ public OffreService(OffreRepository offreRepository, EtudiantRepository etudiant
     public Optional<Offre> saveOffre(Offre offre) {
         return Optional.of(offreRepository.save(offre));
     }
+
+    public Optional<Offre> applyForOffre(int id, String email) {
+        Optional<Offre> offreOptional = offreRepository.findById(id);
+        if (offreOptional.isEmpty()) return offreOptional;
+
+        Optional<Etudiant> etudiantOptional = Optional.ofNullable(etudiantRepository.findEtudiantByCourrielIgnoreCase(email));
+        if (etudiantOptional.isEmpty()) return Optional.empty();
+
+        Offre offre = offreOptional.get();
+        Etudiant etudiant = etudiantOptional.get();
+
+        offre.apply(etudiant);
```

Comment:
Not sure what offre.apply is doing here. Arent we just adding the etudiant to the applicants list in offre?

Reply:
Yes, but since the list of applicants in Offre is private, we need to create a method to add an etudiant.

Otherwise it would look like:
```Set<Etudiant> applicants = offre.getApplicants();
applicants.add(etudiant);
offre.setApplicants(applicants);```

I thought this way would be cleaner